### DISCUSSION TITLE
@ant-design/web3-farcaster

### DISCUSSION THREAD

**Discussion by AmAzing129** [ant-design/ant-design-web3#903] [Ideas]
Farcaster 协议目前受到圈内越来越关注。而相关生态 APP 处于早期起步阶段，一片蓝海。是否可以调研市场需求，提供独立 UI / 工具包？
https://docs.farcaster.xyz/
https://github.com/a16z/awesome-farcaster

**Comment by yutingzhao1991** [comment]
有没有什么具体的想法？

我看官方有一个 https://docs.farcaster.xyz/auth-kit/sign-in-button 这个

是不是我们也可以支持这样类似的组件，可以快速基于 farcster 的协议登录

**Reply by MarioJames** [reply]
现在 Warpcast 的登录体验有点差
PC上面需要扫码唤醒手机端去登录
如果我们可以实现直接 PC 端直接登录的话应该是一个不错的切入点

**Comment by MarioJames** [comment]
目前可以想到的可以做的口子：
1. 每个 Cast 右下角的 Action 插件，可以做一个插件的适配器，快速对接各类底层合约
2. Frame 里面的按钮

**Comment by AmAzing129** [comment]
I created a playground repo and we can do something to see what developers need and how it works. https://github.com/AmAzing129/web3-farcaster-app

**Comment by jeasonstudio** [comment]
简单看了下 `farcaster/auth-kit` 的实现，其中 sign-in 部分好像是 SIWE（[EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)） 的子集：
https://github.com/farcasterxyz/auth-monorepo/blob/main/packages/auth-client/src/messages/build.ts#L17

**Comment by jeasonstudio** [comment]
尝试下添加 warpcast 连接方式到 connect 组件

**Comment by jeasonstudio** [comment]
@MarioJames 跟进下

**Reply by AmAzing129** [reply]
@MarioJames 这个扫码登陆我来开发吧～开始搞了
